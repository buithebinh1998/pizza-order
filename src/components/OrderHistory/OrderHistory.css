#table {
    display:table;
    margin: 0 auto;
    margin-bottom: 2rem;
    border-collapse:collapse;
    font-size: 1rem;
    background-color: whitesmoke;
}

.row  {
    display: table-row;   
}

.row.main{
    color: #04354c;
    font-weight: bold;
}

.cell {
    display: table-cell;
    vertical-align: middle;
    text-align: center;
    border: black solid 1px;
    padding: 0.5rem 0;
}

.cell.id{
    width: 3rem;
}

.cell.orderCart{
    width: 20rem;
}

.cell.totalPrice{
    width: 10rem;
    font-weight: bold;
}

.cell.orderTime{
    width: 10rem;
}

.cell.isDone{
    width: 8rem;
}

.cell.paymentMethod{
    width: 8rem;
}

.cell.address{
    width: 20rem;
}

@media (max-width: 992px) {
    #table {
        font-size: 0.9rem;
    }
    .cell.id{
        width: 2.5rem;
    }
    
    .cell.orderCart{
        width: 15rem;
    }
    
    .cell.totalPrice{
        width: 6rem;
    }
    
    .cell.orderTime{
        width: 10rem;
    }
    
    .cell.isDone{
        width: 5rem;
    }
    
    .cell.paymentMethod{
        width: 5rem;
    }
    
    .cell.address{
        width: 12rem;
    }    

    .cell {
        padding: 0.3rem 0;
    }
}

@media (max-width: 768px) {
    #table {
        font-size: 0.8rem;
    }
    .cell.id{
        width: 2rem;
    }
    
    .cell.orderCart{
        width: 12rem;
    }
    
    .cell.totalPrice{
        width: 5rem;
    }
    
    .cell.orderTime{
        width: 8rem;
    }
    
    .cell.isDone{
        width: 4rem;
    }
    
    .cell.paymentMethod{
        width: 4rem;
    }
    
    .cell.address{
        width: 10rem;
    }    

    .cell {
        padding: 0.3rem 0;
    }
}

@media (max-width: 414px) {
    #table {
        font-size: 0.6rem;
    }

    .cell.id{
        width: 3rem;
    }
    
    .cell.orderCart{
        width: 8rem;
    }
    
    .cell.totalPrice{
        width: 4rem;
    }
    
    .cell.orderTime{
        width: 6rem;
    }
    
    .cell.isDone{
        width: 3rem;
    }
    
    .cell.paymentMethod{
        width: 3rem;
    }
    
    .cell.address{
        width: 8rem;
    }    

    .cell {
        padding: 0.1rem 0;
    }
}

@media (max-width: 320px) {
    #table {
        font-size: 0.5rem;
    }
}