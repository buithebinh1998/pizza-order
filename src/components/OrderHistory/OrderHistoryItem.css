.order-history-cart-item {
  display: flex;
  text-align: left;
  flex-direction: column;
  margin-bottom: 1rem;
  padding: 0 1rem;
}

.order-history-cart-name {
  font-weight: bold;
}

.order-history-cart-total-wrapper {
  display: flex;
  justify-content: flex-start;
}

.order-history-cart-price {
  margin-left: auto;
}

@media (max-width: 992px) {
  .order-history-cart-item {
    margin-bottom: 0.8rem;
    padding: 0 0.8rem;
  }
}
@media (max-width: 768px) {
  .order-history-cart-item {
    margin-bottom: 0.6rem;
    padding: 0 0.6rem;
  }
}
@media (max-width: 414px) {
  .order-history-cart-item {
    margin-bottom: 0.4rem;
    padding: 0 0.4rem;
  }
}
