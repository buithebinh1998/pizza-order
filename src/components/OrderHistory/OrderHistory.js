import React, { useEffect, useState, useContext } from "react";
import "./OrderHistory.css";
import axios from 'axios'
import OrderHistoryItem from "./OrderHistoryItem";
import { Context } from "../../context/Context/Context";

const OrderHistory = () => {
  const [orderHistory, setOrderHistory] = useState();
  const {user} = useContext(Context);

  useEffect(() => {
    if (!orderHistory) {
      axios.get(`https://ec2-52-221-225-178.ap-southeast-1.compute.amazonaws.com:8080/pycozza/order/email?email=${user.email}`)
      .then((response) => {
        setOrderHistory(response.data.orders);
      })
      .catch(error=>{
        console.log(error);
      });
    }
  });

  const showOrderHistory = orderHistory === null ? <h3 className="order-history-message">You haven't ordered anything!</h3> :
    <div id="table">
        <div className="row main">
          <div className="cell id">ID</div>
          <div className="cell orderCart">Cart</div>
          <div className="cell totalPrice">Total Price</div>
          <div className="cell orderTime">Order Time</div>
          <div className="cell address">Address</div>
          <div className="cell paymentMethod">Payment</div>
          <div className="cell isDone">Done</div>
        </div>
        {orderHistory && orderHistory.map((item, index) => (
          <OrderHistoryItem
            key={index}
            id={index}
            cart={item.cart}
            orderTime = {item.orderTime}
            totalPrice={item.totalPrice}
            paymentMethod={item.paymentMethod}
            address={item.user.address}
          />
        ))}
      </div>    
  
  return (
    <div>
      {showOrderHistory}
    </div>
  );
};

export default OrderHistory;
