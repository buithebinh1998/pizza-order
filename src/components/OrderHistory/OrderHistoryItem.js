import React from "react";
import "./OrderHistoryItem.css";
const OrderHistoryItem = (props) => {
  const orderHistoryCart = props.cart;
  const loadOrderHistoryCart = orderHistoryCart.map((item, index) => {
    let showPrice = item.price * 1000;
    let showPriceString = new Intl.NumberFormat("de-DE").format(showPrice);
    return (
      <div className="order-history-cart-item" key={index}>
        <div className="order-history-cart-name">{item.name}</div>
        <div className="order-history-cart-total-wrapper">
          <div>x{item.quantity}</div>

          <div className="order-history-cart-price">
            {showPriceString + "Đ"}
          </div>
        </div>
      </div>
    );
  });

  let showTotalPrice = props.totalPrice * 1000;
  let showTotalPriceString = new Intl.NumberFormat("de-DE").format(
    showTotalPrice
  );
  return (
    <div className="row">
      <div className="cell id">{props.id + 1}</div>
      <div className="cell orderCart">{loadOrderHistoryCart}</div>
      <div className="cell totalPrice">{showTotalPriceString + "Đ"}</div>
      <div className="cell orderTime">{props.orderTime}</div>
      <div className="cell address">{props.address}</div>
      <div className="cell paymentMethod">{props.paymentMethod}</div>
      <div className="cell isDone">Delivered</div>
    </div>
  );
};

export default OrderHistoryItem;
