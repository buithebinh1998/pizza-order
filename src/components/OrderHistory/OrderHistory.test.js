import { testSnapshots } from "../../utils/test.util";

import OrderHistory from "./OrderHistory";

describe("<OrderHistory / >", () => {
  testSnapshots(OrderHistory, [
    {
      props: {},
      description: "default render"
    }
  ]);
});