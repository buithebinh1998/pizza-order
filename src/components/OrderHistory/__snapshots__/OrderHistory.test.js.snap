// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderHistory / > snapshots default render 1`] = `
<div>
  <div
    id="table"
  >
    <div
      className="row main"
    >
      <div
        className="cell id"
      >
        ID
      </div>
      <div
        className="cell orderCart"
      >
        Cart
      </div>
      <div
        className="cell totalPrice"
      >
        Total Price
      </div>
      <div
        className="cell orderTime"
      >
        Order Time
      </div>
      <div
        className="cell address"
      >
        Address
      </div>
      <div
        className="cell paymentMethod"
      >
        Payment
      </div>
      <div
        className="cell isDone"
      >
        Status
      </div>
    </div>
  </div>
</div>
`;
